Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp

```diff
@@ -50,15 +62,36 @@ FreeVector estimateTrackParamsFromSeed(const Vector3& sp0, const Vector3& sp1,
 
 /// Estimate the full track parameters from three space points
 ///
-/// This method is based on the conformal map transformation. It estimates the
-/// full free track parameters, i.e. (x, y, z, t, dx, dy, dz, q/p) at the bottom
-/// space point. The bottom space is assumed to be the first element in the
-/// range defined by the iterators. The magnetic field (which might be along any
-/// direction) is also necessary for the momentum estimation.
+/// This is a purely spatial estimation, i.e. the time parameter will be set to
+/// 0.
 ///
-/// It resembles the method used in ATLAS for the track parameters estimated
-/// from seed, i.e. the function InDet::SiTrackMaker_xk::getAtaPlane here:
-/// https://acode-browser.usatlas.bnl.gov/lxr/source/athena/InnerDetector/InDetRecTools/SiTrackMakerTool_xk/src/SiTrackMaker_xk.cxx
+/// See the free parameter version for more details.
+///
+/// @tparam spacepoint_iterator_t  The type of space point iterator
+///
+/// @param gctx the geometry context
+/// @param mctx the magnetic field context
+/// @param surface the surface where the bound parameters should be expressed
+/// @param sp0 the bottom space point
+/// @param sp1 the middle space point
+/// @param sp2 the top space point
+/// @param bField the magnetic field provider
+/// @param propagator is the propagator used in case the bottom spacepoint is not on the surface
+/// @param logger the logger to be used
+///
+/// @return bound parameters result
+Result<BoundVector> estimateTrackParamsFromSeedAtSurface(
+    const GeometryContext& gctx, const MagneticFieldContext& mctx,
+    const Surface& surface, const Vector3& sp0, const Vector3& sp1,
+    const Vector3& sp2,
+    const std::shared_ptr<const MagneticFieldProvider>& bField,
```

Comment:
You need this as shared to be able to construct the propagator, right?

Reply:
Yes the stepper really wants have a shared pointer. I thought it might be good to pass in the shared pointer as a reference so we avoid a copy in case the user also provides a propagator.

We will pay the overhead on the propagator construction on every call if it is constructed by us though. I guess it is not huge but in principle avoidable by factorizing this function in estimate, transform, propagate.